### ISSUE TITLE
Issue with "XMLHttpRequest": …

### ISSUE THREAD

**Issue by joopringelberg** [mdn/content#1515]
MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest

#### What information was incorrect, unhelpful, or incomplete?
The text claims: "Note:This feature is available in Web Workers." but I don't think it is.

#### Specific section or headline?
The very first section.

#### What did you expect to see?
Note:This feature is **not** available in Web Workers.

#### Did you test this? If so, how?
By running a service worker with a new XMLHttpRequest() expression in it.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/api/xmlhttprequest`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest
* GitHub URL: https://github.com/mdn/content/blob/master/files/en-us/web/api/xmlhttprequest/index.html
* Report started: 2021-01-19T21:41:59.172Z

</details>

**Comment by chrisdavidmills** [comment]
I think you are right here; removed via https://github.com/mdn/content/commit/b0e464227a76845087bcd246f6f5ca6298f23dc1

Thanks for reporting.

**Comment by hamishwillee** [comment]
@chrisdavidmills @joopringelberg WHY don't you think it is supported in web workers?

I tested on FF and Chrome using https://worker-playground.glitch.me/ - that shows XMLHttpRequest is defined in Shared/Dedicated workers, but not Service Workers

https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API says it is.
![image](https://user-images.githubusercontent.com/5368500/105441579-fb9e6180-5cbc-11eb-9096-d44c338b38c4.png)

And in fact I would say you should **only** use these in web workers.

**Comment by joopringelberg** [comment]
You must be right. I encountered a problem on using a library that relies on XmlHttpRequest, running it in a *ServiceWorker* context. I did not test the other worker types and jumped to the conclusion that this API is not supported by workers in general. Even though the terminology is exact ('workers', 'web-workers' and 'service workers) I find myself treating the part (web-workers) for the whole (workers).
To sum up: the MDN text could be more helpful in stating: This feature is available in Web Workers (but not in other types of Worker)

**Comment by chrisdavidmills** [comment]
@hamishwillee I don't know what I was thinking here ;-)

**Comment by joopringelberg** [comment]
I must retract 'Even though the terminology is exact'. The [Web Workers API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API) introduces 'workers' as a general concept in the paragraph 'Web Workers concepts and usage' (so it had better be titled 'Worker concepts and usage'. It states: 'In addition, workers may use XMLHttpRequest for network I/O (...).
 It then goes on to distinguish worker types, listing dedicated workers, shared workers, service workers and Chrome workers (notice web workers are absent!).
So this page seems to equate 'web worker' with the category 'worker', while the page that I originally commented on uses the term for the subcategory.

**Comment by joopringelberg** [comment]
I must retract 'Even though the terminology is exact'. The [Web Workers API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API) introduces 'workers' as a general concept in the paragraph 'Web Workers concepts and usage' (so it had better be titled 'Worker concepts and usage'. It states: 'In addition, workers may use XMLHttpRequest for network I/O (...).
 It then goes on to distinguish worker types, listing dedicated workers, shared workers, service workers and Chrome workers (notice web workers are absent!).
So this page seems to equate 'web worker' with the category 'worker', while the page that I originally commented on uses the term for the subcategory.

**Comment by hamishwillee** [comment]
@joopringelberg Yes :-). So the way we plan on handling this going forward is using the compatibility data to indicate things that work in service workers, and those that don't.

**Comment by hamishwillee** [comment]
No worries. I've [reverted the change](https://github.com/mdn/content/pull/1700). BCD update [here](https://github.com/mdn/browser-compat-data/pull/8853). I will revisit the content once the BCD is confirmed. In the meantime will leave this open.

**Comment by hamishwillee** [comment]
OK, so BCD is updated to make it clear where different service workers are updated. I also added a note up top in #2185.

Closing now. Once those goes live this will be fixed.
